src/Features/LanguageServer/Protocol/Extensions/Extensions.cs

```diff
@@ -77,23 +77,18 @@ public static ImmutableArray<DocumentId> GetDocumentIds(this Solution solution,
             // For now we do our best to handle as many cases as we can.
             // Tracking issue - https://github.com/dotnet/roslyn/issues/68083
 
-            // If the uri is a file then use the simplified absolute or local path.
-            // In other cases (e.g. git files) use the full uri string. This ensures documents
-            // with the same local path (e.g. git://someFilePath and file://someFilePath) are differentiated.
-            if (documentUri.IsFile)
-            {
-                var fileDocumentIds = solution.GetDocumentIdsWithFilePath(documentUri.AbsolutePath);
-                if (fileDocumentIds.Any())
-                    return fileDocumentIds;
+            // Use the original string the Uri was created with to avoid any Unicode escaping issues
+            // (see https://dev.azure.com/devdiv/DevDiv/_workitems/edit/1842233).
+            // Only if the original string is not an existing document path and the Uri has "file" scheme,
+            // use the LocalPath, in case the client creates Uri instance with using the URL string "file://...".
 
-                fileDocumentIds = solution.GetDocumentIdsWithFilePath(documentUri.LocalPath);
```

Comment:
If you are otherwise happy with the behavior of `LocalPath` (that is, escaping the PUA characters is the only issue), have you considered using `Uri.UnescapeDataString(documentUri.LocalPath)` instead?

Reply:
I don't think we should escape anything. I guarantee that any code that does any escaping like that is wrong.

Reply:
```C#
> Uri.UnescapeDataString(new Uri("C:\\\ue25b%25").LocalPath)
"C:\\%"
```